[mlir][vector] Add more tests for ConvertVectorToLLVM (8/n)

banach-space opened the pull request:
Adds tests with scalable vectors for the Vector-To-LLVM conversion pass.
Covers the following Ops:
  * vector.transfer_read
  * vector.transfer_write

In addition:
  * Duplicate tests from "vector-mask-to-llvm.mlir" are removed.
  * Tests for xfer_read/xfer_write are moved to a newly created test file,
     "vector-xfer-to-llvm.mlir". This follows an existing pattern among
     VectorToLLVM conversion tests.
  * Tests that test both xfer_read and xfer_write have their names
    updated to capture that (e.g. `@transfer_read_1d_mask` ->
    `@transfer_read_write_1d_mask`)
  * `@transfer_write_1d_scalable_mask` and
    `@transfer_read_1d_scalable_mask` are re-written as
    `@transfer_read_write_1d_mask_scalable`. This is to make it clear
    that this case is meant to complement
    `@transfer_read_write_1d_mask`.
  * `@transfer_write_tensor` is updated to also test `xfer_read`.

banach-space committed:
[mlir][vector] Add more tests for ConvertVectorToLLVM (8/n)

Adds tests with scalable vectors for the Vector-To-LLVM conversion pass.
Covers the following Ops:
  * vector.transfer_read
  * vector.transfer_write

In addition:
  * Tests that test both xfer_read and xfer_write have their names
    updated to capture that (e.g. `@transfer_read_1d_mask` ->
    `@transfer_read_write_1d_mask`)
  * `@transfer_write_1d_scalable_mask` and
    `@transfer_read_1d_scalable_mask` are re-written as
    `@transfer_read_write_1d_mask_scalable`. This is to make it clear
    that this case is meant to complement
    `@transfer_read_write_1d_mask`.
  * `@transfer_write_tensor` is updated to also test `xfer_read`.

javedabsar1 commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -2449,9 +2449,77 @@ func.func @transfer_read_1d(%A : memref<?xf32>, %base: index) -> vector<17xf32>
+//  CHECK-SAME: %[[BASE:.*]]: index) -> vector<[17]xf32>
+//       CHECK: %[[C7:.*]] = arith.constant 7.0
+//
+// 1. Let dim be the memref dimension, compute the in-bound index (dim - offset)
```
I like this commenting style of explaining sections of CHECKS.

javedabsar1 commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -2449,9 +2449,77 @@ func.func @transfer_read_1d(%A : memref<?xf32>, %base: index) -> vector<17xf32>
+//
+// 1. Let dim be the memref dimension, compute the in-bound index (dim - offset)
+//       CHECK: %[[C0:.*]] = arith.constant 0 : index
+//       CHECK: %[[DIM:.*]] = memref.dim %[[MEM]], %[[C0]] : memref<?xf32>
```
nit. commiting to tab space before '       CHECK' reduces readability for long lines unnecessarily. maybe reduce to '  CHECK'.

javedabsar1 commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -2377,7 +2377,7 @@ func.func @matrix_ops_index(%A: vector<64xindex>, %B: vector<48xindex>) -> vecto
 // -----
 
-func.func @transfer_read_1d(%A : memref<?xf32>, %base: index) -> vector<17xf32> {
+func.func @transfer_read_write_1d(%A : memref<?xf32>, %base: index) -> vector<17xf32> {
```
This file has become too big. Would you consider separating 'vector-transer-to-llvm.mlir like existing vector-mask-to-llvm ?

banach-space committed:
fixup! [mlir][vector] Add more tests for ConvertVectorToLLVM (8/n)

Create a dedicated file for xfer tests

banach-space committed:
fixup! fixup! [mlir][vector] Add more tests for ConvertVectorToLLVM (8/n)

Add missing tests in vector-xfer-to-llvm.mlir, remove duplicates from vector-mask-to-llvm.mlir

javedabsar1 reviewed (approved):
LGTM. Thanks.
